### PR TITLE
version check changes

### PR DESCRIPTION
**Pull request by rakeshsharma14317**

### CODE REVIEW

File: cli/pkg/update/update.go [hunk 1]
```diff
@@ -64,12 +65,12 @@ func LatestVersion(ctx context.Context) (string, error) {
 			return "", err
 		}
 
-		updatedAt, err := time.Parse("2006-01-02 15:04", cachedVersionUpdatedAt)
+		updatedAt, err := time.Parse(cmdutil.TSFormatLayout, cachedVersionUpdatedAt)
```
Comment (begelundmuller) [inline cli/pkg/update/update.go:68]: This should not use the user-facing timestamp format, it's for internal persistance. Use `time.RFC3339` instead

File: cli/pkg/update/update.go [hunk 2]
```diff
@@ -64,12 +65,12 @@ func LatestVersion(ctx context.Context) (string, error) {
...
 		if err != nil {
 			return "", err
 		}
 
-		if time.Since(updatedAt).Hours() < versionCheckTTL.Hours() {
+		if time.Since(updatedAt.UTC()).Hours() < versionCheckTTL.Hours() {
```
Comment (begelundmuller) [inline cli/pkg/update/update.go:73]: 1. `UTC` will not be needed here when using `time.RFC3339`
2. Don't do `Hours()`, just compare the time deltas

File: cli/pkg/update/update.go [hunk 3]
```diff
@@ -80,7 +81,7 @@ func LatestVersion(ctx context.Context) (string, error) {
 		return "", err
 	}
 
-	err = dotrill.SetVersionUpdatedAt(time.Now().Format("2006-01-02 15:04"))
+	err = dotrill.SetVersionUpdatedAt(time.Now().UTC().Format(cmdutil.TSFormatLayout))
```
Comment (begelundmuller) [inline cli/pkg/update/update.go:84]: Don't do `UTC` and use `time.RFC3339`

File: cli/cmd/root.go [hunk 4]
```diff
@@ -87,7 +87,7 @@ func runCmd(ctx context.Context, ver config.Version) error {
 	// Check version
 	err := update.CheckVersion(ctx, cfg.Version.Number)
 	if err != nil {
-		return err
+		fmt.Printf("check version failed, Error: %v\n", err)
```
Comment (begelundmuller) [inline cli/cmd/root.go:90]: Make it `Warning: version check failed: %v`

**Review by begelundmuller — CHANGES_REQUESTED** [review]

File: cli/pkg/update/update.go (hunk 1 above)
Comment (rakeshsharma14317) [reply cli/pkg/update/update.go:68]: Thanks, i was checking for some better approach for this, But yeah this looks good for internal persistance :)

**Review by begelundmuller — APPROVED** [review]

**Pull request merged**